src/vsr/replica.zig

```diff
@@ -5798,7 +5840,14 @@ pub fn ReplicaType(
         /// doing so would overwrite a message (or the slot of a message) that has not yet been
         /// committed and checkpointed.
         fn op_prepare_max(self: *const Self) u64 {
-            return vsr.Checkpoint.prepare_max_for_checkpoint(self.op_checkpoint_next()).?;
+            const checkpoint = checkpoint: {
+                if (self.checkpoint_quorum) {
+                    break :checkpoint self.op_checkpoint_next();
+                } else {
+                    break :checkpoint self.op_checkpoint();
+                }
+            };
```

Comment:
Don't need a block here:

```suggestion
            const checkpoint = if (self.checkpoint_quorum) 
                    self.op_checkpoint_next()
                else 
                    self.op_checkpoint();
```